src/contacts/store/index.js

```diff
@@ -0,0 +1,114 @@
+import {
+  getContacts,
+  updateContactStatus,
+  deleteContact,
+} from "@/http/contacts.js";
+import Vue from "vue";
+import NProgress from "nprogress";
+const namespaced = true;
+const state = {
+  contacts: [],
+  pagination: {
+    pageNumber: 1,
+    totalPages: 0,
+    pageSize: 10,
+  },
+};
+const mutations = {
+  SET_CONTACTS(state, data) {
+    state.contacts = data;
+  },
+  SET_PAGINATION(state, pagination) {
+    state.pagination = {
+      ...state.pagination,
+      ...pagination,
+    };
+  },
+  SET_CONTACT_STATUS(state, { contactId, contactStatus, contactTypeChange }) {
+    const index = state.contacts.findIndex((c) => c.id === contactId);
+    if (contactTypeChange === 0)
+      state.contacts[index].customerStatus = contactStatus;
+    else state.contacts[index].supplierStatus = contactStatus;
```

Comment:
We should have `{}` for all `if else` statements, it's a javascript convention.
```suggestion
    if (contactTypeChange === 0) {
      state.contacts[index].customerStatus = contactStatus;
    } else {
      state.contacts[index].supplierStatus = contactStatus;
    }
```